File: frontend/src/types/template.ts
export interface TemplateField {
  name: string;
  label: string;
  type: string;
  required: boolean;
  source: "1c" | "manual" | "ai";
}

export interface Template {
  id: string;
  slug: string;
  title: string;
  description: string;
  department: string;
  model_name: string;
  fields: TemplateField[];
  is_active: boolean;
}

export interface TemplateMeta {
  id: string;
  title: string;
  department: string;
  model: string;
  fields: TemplateField[];
}

File: backend/app/schemas/template.py
from pydantic import BaseModel
from typing import Optional
import uuid


class FieldSchema(BaseModel):
    name: str
    label: str
    type: str = "text"
    required: bool = False
    source: str = "manual"


class TemplateResponse(BaseModel):
    id: uuid.UUID
    slug: str
    title: str
    description: str
    department: str
    model_name: str
    fields: list[FieldSchema]
    is_active: bool

    model_config = {"from_attributes": True}


class TemplateCreate(BaseModel):
    slug: str
    title: str
    description: str = ""
    department: str
    model_name: str = "llama3.1:8b"
    fields: list[FieldSchema] = []
    file_path: str
